dashboard/app/models/teacher_feedback.rb

```diff
@@ -57,23 +60,45 @@ def get_script_level
     script_level
   end
 
-  def self.get_student_level_feedback(student_id, level_id, teacher_id, script_id)
+  def self.get_student_level_feedback_by_teacher(student_id, level_id, script_id, teacher_id)
```

Comment:
Maybe this could just be `get_latest_feedback`?  Also, somewhat minor, but it would be good to pass these parameters in a consistent order (and maybe even pass by name since it could be easy to confuse them).